### PR TITLE
feat: auth and archive

### PR DESCRIPTION
**Pull request by dev-ashishsethi**
https://user-images.githubusercontent.com/89299065/166310918-045669eb-7adf-49c9-adf1-fe1f00f5b2b5.mp4

image

Features:

SignUp
Log Out
Archive
Un-Archive

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -19,7 +21,9 @@ function App() {
       <Routes>
         <Route path="/" element={<LandingPage />} />
         <Route path="/signIn" element={<SignIn />} />
+        <Route path="/signUp" element={<SignUp />} />
         {login && <Route path="/home" element={<Home />} />}
+        {login && <Route path="/archive" element={<Archive />} />}
```
Comment (tarunsankhla) [inline src/App.js:26]: Can create path of constant

File: src/Components/SignUp/SignUp.jsx [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+        return { ...state, email: action.email };
+      case "Password":
+        return { ...state, password: action.password };
+      case "ConfirmPassword":
+        return { ...state, confirmPassword: action.confirmPassword };
+    }
+  }
+  const showPassword = () =>
+    passwordType === "password"
+      ? setPasswordType("text")
```
Comment (tarunsankhla) [inline src/Components/SignUp/SignUp.jsx:36]: Can create const of these strings too

File: src/Components/SignUp/SignUp.jsx [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+      : setPasswordType("password");
+
+  const submitHandler = async (e) => {
+    e.preventDefault();
+    if (
+      state.firstName !== "" &&
```
Comment (tarunsankhla) [inline src/Components/SignUp/SignUp.jsx:42]: Do directly !state.firstName

File: src/Components/SignUp/signUp.css [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    "fname lname"
+    "finput linput"
+    "emaillabel passlabel"
+    "email pass"
+    "confpasslabel -"
+    "confpass -"
+    "btn1 btn2";
+
+  grid-template-columns: 1fr 1fr;
+  grid-template-rows: 0fr 0fr 0fr 0fr 0fr 0fr;
```
Comment (tarunsankhla) [inline src/Components/SignUp/signUp.css:13]: I think fr is not required use directly 0

File: src/Components/SignUp/signUp.css [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+.form-container {
+  display: flex;
+  place-content: center;
+  align-items: center;
+  height: 83vh;
+}
+
+.form-subcontainer {
+  padding: 2rem 4rem 4rem 4rem;
```
Comment (tarunsankhla) [inline src/Components/SignUp/signUp.css:77]: No need of last 4 rem

File: src/Pages/Archive/Archive.jsx [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+      try {
+        const response = await axios({
+          method: "POST",
+          url: `/api/archives/restore/${noteId}`,
+          data: { note: note },
+          headers: { authorization: token },
+        });
+        setArchiveNotes(response.data.archives);
+        setOthersData(response.data.notes);
+        console.log(response);
```
Comment (tarunsankhla) [inline src/Pages/Archive/Archive.jsx:21]: Remove console

File: src/index.js [hunk 7]
```diff
@@ -25,3 +26,8 @@ ReactDOM.render(
   </React.StrictMode>,
   document.getElementById("root")
 );
+
+// ReactDOM.render(
+//   ,
```
Comment (tarunsankhla) [inline src/index.js:31]: Remove comments

**Review by tarunsankhla — APPROVED** [review]
Overall great minor changes

**Review by m26unkwn — APPROVED** [review]
Remove unwanted comments

and try to extract all the async func into one file .

File: src/App.js [hunk 8]
```diff
@@ -8,6 +8,8 @@ import { Route, Routes } from "react-router-dom";
 import { SignIn } from "./Components/SignIn/SignIn";
 import { LandingPage } from "./Pages/LandingPage/LandingPage";
 import { useAuth } from "./Context/loginContext";
+import { SignUp } from "./Components/SignUp/SignUp";
+import { Archive } from "./Pages/Archive/Archive";
```
Comment (SinchanaMS) [inline src/App.js:12]: you could import from a single index file for common paths - components, and pages.

File: src/Components/SignUp/signUp.css (hunk 4 above)
Comment (SinchanaMS) [inline src/Components/SignUp/signUp.css:13]: you could just use `grid-template-rows: repeat(no-of-rows, row-width)` if the number of rows is fixed.

**Review by SinchanaMS — APPROVED** [review]
Have provided a few minor suggestions. 
Also, You can move the async functions to a separate file.

**Pull request merged**